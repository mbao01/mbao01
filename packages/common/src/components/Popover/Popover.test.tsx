import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it } from "vitest";
import { Popover } from "./";

describe("Popover", () => {
  const renderPopover = () =>
    render(
      <Popover>
        <Popover.Trigger>Open popover</Popover.Trigger>
        <Popover.Content className="w-80">Here is some easter egg for you ðŸ¥š</Popover.Content>
      </Popover>
    );

  it("renders a popover", async () => {
    const user = userEvent.setup();
    const { asFragment } = renderPopover();

    const popoverTrigger = screen.getByRole("button", { name: "Open popover" });

    expect(screen.queryByText("Here is some easter egg for you ðŸ¥š")).not.toBeInTheDocument();

    await user.click(popoverTrigger);

    expect(screen.getByText("Here is some easter egg for you ðŸ¥š")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
