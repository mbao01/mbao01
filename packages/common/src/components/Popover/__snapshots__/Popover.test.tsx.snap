// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popover > renders a popover 1`] = `
<DocumentFragment>
  <button
    aria-controls="radix-«r0»"
    aria-expanded="true"
    aria-haspopup="dialog"
    data-state="open"
    type="button"
  >
    Open popover
  </button>
</DocumentFragment>
`;
